Proton/Sources/Editor/EditorView.swift

```diff
@@ -231,7 +231,7 @@ open class EditorView: UIView {
     /// An attachment is only counted as a single character. Content length does not include
     /// length of content within the Attachment that is hosting another `EditorView`.
     public var contentLength: Int {
-        return attributedText.length
+        return richTextView.contentLength
```

Comment:
This is not related to substrings, but is related to allocating new `NSAttributedString` every time we want the `contentLength`. This property is called from `Attachment.attachmentBounds(for: ...)`, and that means it gets called a lot, and that means we're allocating a lot of NSAttributedStrings for nothing!